CSharp/Library/Microsoft.Bot.Connector.Shared/MicrosoftAppCredentials.cs

```diff
@@ -83,7 +83,18 @@ public MicrosoftAppCredentials(IConfiguration configuration, ILogger logger = nu
         public string MicrosoftAppId { get; set; }
         public string MicrosoftAppPassword { get; set; }
 
-        public virtual string OAuthEndpoint { get { return JwtConfig.ToChannelFromBotLoginUrl; } }
+        public virtual string OAuthEndpoint
+        {
+            get
+            {
+                string tenant = null;
+#if NET45
+                // Advanced user only. TODO: add doc link.
+                tenant = SettingsUtils.GetAppSettings("ChannelAuthTenant").NullIfEmpty();

```

Comment:
Given that this is the only place the Extension method is used, I would delete it and use a simple if/else here. That will also make porting this to JS/Java/Python easier, as the code will be exactly the same.